tests/e2e/utils/reset.js

```diff
@@ -32,21 +32,37 @@ import {
 
 /**
  * Resets Site Kit using a utility plugin.
+ *
+ * @since 1.0.0
+ * @since n.e.x.t Option to reset persistent options.
+ *
+ * @param {Object} options Reset Site kit Options.
  */
-export async function resetSiteKit() {
+export async function resetSiteKit( options = { persistent: false } ) {
 	if ( ! page.url().includes( '/wp-admin' ) ) {
 		await visitAdminPage( '/' );
 	}
 
-	await Promise.all( [
+	const promises = [
 		wpApiFetch( {
 			path: 'google-site-kit/v1/core/site/data/reset',
 			method: 'post',
 		} ),
 		clearLocalStorage(),
 		clearSessionStorage(),
 		page.waitForResponse( ( res ) => res.url().match( 'google-site-kit/v1/core/site/data/reset' ) ),
-	] );
+	];
+
+	if ( options.persistent ) {
+		promises.push(
+			wpApiFetch( {
+				path: 'google-site-kit/v1/core/site/data/reset-persistent',
+				method: 'post',
+			} )
```

Comment:
Let's also wait for the response as well like we do with the main reset above.
```suggestion
			} ),
			page.waitForResponse( ( res ) => res.url().match( 'google-site-kit/v1/core/site/data/reset-persistent' ) )
```